sdk/src/Services/S3/Custom/Model/GetObjectResponse.cs

```diff
@@ -910,6 +910,59 @@ private void ValidateWrittenStreamSize(long bytesWritten)
         }
 
 #if BCL || NETSTANDARD
+        /// <summary>
+        /// Copies data from ResponseStream to destination stream with progress tracking and validation.
+        /// Internal method to enable reuse across different download scenarios.
+        /// </summary>
+        /// <param name="destinationStream">Stream to write data to</param>
+        /// <param name="filePath">File path for progress event reporting (can be null)</param>
+        /// <param name="bufferSize">Buffer size for reading/writing operations</param>
+        /// <param name="cancellationToken">Cancellation token</param>
+        /// <param name="validateSize">Whether to validate copied bytes match ContentLength</param>
+        internal async System.Threading.Tasks.Task WriteResponseStreamAsync(
```

Comment:
this function should be identical to the old one. ive just added this as a helper function for writing to a response stream instead of a file